src/helpers/__tests__/apiHelpers.test.ts

```diff
@@ -0,0 +1,50 @@
+import apiHelpers from '../apiHelpers';
+
+describe('extractErrorMessage', () => {
+  it('should concatenate error messages from an object', () => {
+    const data = {
+      error1: 'Error message 1',
+      error2: 'Error message 2'
+    };
+    expect(apiHelpers.extractErrorMessage(data)).toMatchSnapshot('concatenate error messages from an object');
+  });
+
+  it('should handle arrays of error messages', () => {
+    const data = {
+      errors: ['Error message 1', 'Error message 2']
+    };
+    expect(apiHelpers.extractErrorMessage(data)).toMatchSnapshot('handle arrays of error messages');
+  });
+
+  it('should handle nested arrays of error messages', () => {
+    const data = {
+      errors: ['Error message 1', ['Error message 2', 'Error message 3']]
+    };
+    expect(apiHelpers.extractErrorMessage(data)).toMatchSnapshot('handle nested arrays of error messages');
+  });
+
+  it('should filter out falsy values', () => {
+    const data = {
+      error1: 'Error message 1',
+      error2: null,
+      error3: undefined,
+      error4: '',
+      error5: 'Error message 2'
+    };
+    expect(apiHelpers.extractErrorMessage(data)).toMatchSnapshot('filter out falsy values');
+  });
+
+  it('should return "Unknown error" if no error messages are found', () => {
+    expect(apiHelpers.extractErrorMessage(undefined)).toMatchSnapshot('return "Unknown error" for undefined input');
+  });
+
+  it('should handle objects with mixed types of values', () => {
+    const data = {
+      error1: 'Error message 1',
+      error2: ['Error message 2', false, 0, 'Error message 3'],
+      error3: null,
+      error4: { nestedError: 'Nested error message' }
+    };
+    expect(apiHelpers.extractErrorMessage(data)).toMatchSnapshot('handle objects with mixed types of values');
+  });
+});
```

Comment:
It's still hard for me to choose the most simplistic path, mainly when it comes to tests. I chose the second snippet you proposed above because I feel it ha s abit more granulity compared to the first one, so it gives me a clearer insight on what's being tested, but still much more concise than the tests I wrote.